include/nanorange/detail/algorithm/return_types.hpp

```diff
@@ -0,0 +1,171 @@
+// nanorange/detail/algorithm/return_types.hpp
+//
+// Copyright (c) 2020 Boris Staletic (boris dot staletic at gmail dot com)
```

Comment:
If the implementations here were based on/renamed from my earlier `*_result` implementations (and it looks like they were), then please add my name to the copyright header too :)

Reply:
Done.